### PR TITLE
feat: Update network controller v^10

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
This PR aims to update network controller to version ^10.0.0 as part of the controllers upgrade to version v55

This PR also removes the existing patch, since it's no longer needed.
Also, it updates the assert package, because couldn't find `notStrictEqual` on the old version.

**This PR is blocked until the v53 last [PR](https://github.com/MetaMask/metamask-mobile/pull/7999) is merged, will be a draft until then**

## **Related issues**

Fixes:

## **Manual testing steps**
1. Add/switch/remove networks from our popular list
2. Add/switch/remove manually added
3. Add/switch networs via Dapp

## **Screenshots/Recordings**
https://recordit.co/Mrnu9csKXG - Adding, Switching and removing network between networks on the popular list
https://recordit.co/Bmhhi1eOeG - Adding and Switching network from Dapps and Import it custom manually  

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

## **CI Related Section: DO NOT EDIT**

Flag used by Bitrise to either run or skip E2E smoke tests

- Bitrise Flag:[skip ci]

### CODE REVIEW

**Comment by tommasini** [comment]
e2e smoke pipeline: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/932c8b82-d4ad-4751-b698-0b95e3e53f02

**Review by MarioAslau — DISMISSED** [review]
Looks Good!

File: package.json [hunk 1]
```diff
@@ -443,7 +443,7 @@
     "@wdio/spec-reporter": "^7.19.1",
     "appium": "^1.22.3",
     "appium-adb": "^9.11.4",
-    "assert": "1.4.1",
```
Comment (tommasini) [inline package.json:446]: couldn't find notStrictEqual on the old version.

**Review by georgewrmarshall — APPROVED** [review]
Approving as code owner to unblock merge but have not tested this in any way. Please get another review from an engineer with context of these changes

**Review by MarioAslau — APPROVED** [review]

**Pull request merged**